### PR TITLE
Remove empty expressions from the dynamite request

### PR DESCRIPTION
**Pull request by nicks**
Hello @kylehg, @xiao, 

Please review the following commits I made in branch 'nicks/empty'.

f666fa85e23cc30766eb99879a6eb270506fd370 (2016-08-26 12:05:47 -0400)
Remove empty expressions from the dynamite request

R=@kylehg
R=@xiao

### CODE REVIEW

File: lib/DynamoRequest.js [hunk 1]
```diff
@@ -260,6 +261,15 @@ DynamoRequest.prototype.build = function () {
     delete this.data.ExpressionAttributeValues
   }
 
+  // Dynamo doesn't like it when conditions are empty
+  if (!this.data.KeyConditionExpression) {
```
Comment (xiao) [inline lib/DynamoRequest.js]: If `KeyConditionsExpression` is missing, we have a bigger problem.

Comment (xiao) [inline lib/DynamoRequest.js]: nvm, there are requests that don't need `KeyConditionsExpression`.

**Comment by xiao** [comment]
LGTM

File: lib/DynamoRequest.js (hunk 1 above)
Comment (nicks) [inline lib/DynamoRequest.js]: ya i think if that happens it'll get caught elsewhere (?)

Comment (xiao) [inline lib/DynamoRequest.js]: It will either be caught by the user of dynamite, or DynamoDB will throw hard errors on it.

**Pull request merged**